biapol_utilities/label/_matching_algorithms.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+
+def max_similarity(label_image_x, label_image_y, similarity_matrix):
+    """
+    Matches labels in two input label images (label_image_x and label_image_y) 
+    based on the maximal value in the similarity_matrix.
+
+    Parameters
+    ----------
+    label_image_x : ND-array, int
+        labelled input image which serves as reference image
+    label_image_y : ND-array, int
+        Labbelled input image, the labels of which will be matched up with the 
+        labels in label_image_x based on the chosen input metric
+    metric_matrix : 2D-array, float
+        matrix containing the metric that defines the similarity of labels
+        in label_image_x and label_image_y, with axis=0 referring to the labels 
+        of label_image_x. In other words, the entry similarity_matrix[i, j] corresponds
+        to the "similarity" between label i in label_image_x and label j in
+        label_image_y
+
+    Returns
+    -------
+    ND-array
+        Relabelled version of input_image_y. Unmatched labels are appended to the set 
+        of matched labels.
+
+    """
+    
+    # Suppress non-maximal entries
+    similarity_matrix[similarity_matrix < similarity_matrix.max(axis=0)] = 0.0
+    
+    mmax = label_image_x.max()
+    
+    if similarity_matrix.size > 0:
```

Comment:
What if this is not true?

Reply:
I added a raised `ValueError` in case a zero-sized matrix is passed.